### PR TITLE
update SuiteSparse to 5.1.2-p0

### PR DESCRIPTION
**Pull request by NeroBurner**
- suitesparse-metis-for-windows version: 1.4.0
- SuiteSparse version: 5.1.2
- add additional licence file

* I've followed [this guide](https://docs.hunter.sh/en/latest/creating-new/update.html)
  step by step carefully. **[Yes]**

* I've tested this package remotely and have excluded all broken builds.
  Here is the links to the Travis/AppVeyor with status "All passed":

  * https://ci.appveyor.com/project/NeroBurner/hunter/builds/20269040
  * https://travis-ci.org/NeroBurner/hunter/builds/454564597

* Toolchains are updated by merging of pkg.template
  https://github.com/ingenue/hunter/pull/289

---

### CODE REVIEW

File: cmake/projects/SuiteSparse/hunter.cmake [hunk 1]
```diff
@@ -35,13 +35,23 @@ hunter_add_version(
     SHA1
     1350eab5d1a8d488e7b28d217fa751906d4767da
 )
+hunter_add_version(
+    PACKAGE_NAME
+    SuiteSparse
+    VERSION
+    "5.1.2-p0"
```
Comment (ruslo) [inline cmake/projects/SuiteSparse/hunter.cmake]: Should it be `1.4.0-p0`?

Comment (ruslo) [reply cmake/projects/SuiteSparse/hunter.cmake]: > suitesparse-metis-for-windows version: 1.4.0
> SuiteSparse version: 5.1.2

Okay, I see

**Comment by NeroBurner** [comment]
I have a problem with the additional license file. When I add it the build fails (and I was not able to find a good error message in the build-output)
- failing travis builds: https://travis-ci.org/NeroBurner/hunter/builds/454849681

When I remove the additional license the test branch builds successfully
- successful travis builds: https://travis-ci.org/NeroBurner/hunter/builds/454889764

What is the format to add multiple license files? Or is that not supported?

When I update the license file I'll also add a comment about the suitesparse-metis and SuiteSparse version difference

**Comment by ruslo** [comment]
> What is the format to add multiple license files? Or is that not supported?

Should work fine: https://github.com/ruslo/hunter/blob/v0.23.51/scripts/try-copy-license.cmake#L28-L42

> I have a problem with the additional license file. When I add it the build fails (and I was not able to find a good error message in the build-output)

Can you try to run it locally so you can check log files? Does `SuiteSparse_licenses.txt` present in old versions?

**Comment by NeroBurner** [comment]
I'm so stupid! :see_no_evil: 
The additional license file is only locally on my PC, it was never in the repository! Thanks for the testing and sorry for the noise. Will fix shortly

**Comment by ruslo** [comment]
> The additional license file is only locally on my PC, it was never in the repository!

The benefits of having remote testing :)

**Comment by NeroBurner** [comment]
updated. Shall I do the `test.suitesparse` CI compilation again?

**Comment by ruslo** [comment]
> Shall I do the test.suitesparse CI compilation again?

No, I've launched tests:
* https://ci.appveyor.com/project/ingenue/hunter/builds/20293646
* https://travis-ci.org/ingenue/hunter/builds/454969629?utm_source=github_status&utm_medium=notification

But in general you should not open pull request if tests are not working:

> I've tested this package remotely and have excluded all broken builds.
> Here is the links to the Travis/AppVeyor with status **"All passed"**

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/builds/20296714
* https://travis-ci.org/ingenue/hunter/builds/455015604?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.23.52

**Pull request merged**